Build fails for Clang 6.0, Ubuntu, in pybindings SDR_MetricsHelper_

I get the following error when building the py bindings on clang-6.0 
```
86%] Building CXX object bindings/py/cpp_src/CMakeFiles/algorithms.dir/bindings/algorithms/py_SDR_Metrics.cpp.o
In file included from /home/mmm/devel/HTM/htm-community/nupic.cpp/bindings/py/cpp_src/bindings/algorithms/py_SDR_Metrics.cpp:22:
In file included from /home/mmm/devel/HTM/htm-community/nupic.cpp/build/ThirdParty/pybind11/pybind11-src/include/pybind11/pybind11.h:43:
In file included from /home/mmm/devel/HTM/htm-community/nupic.cpp/build/ThirdParty/pybind11/pybind11-src/include/pybind11/detail/../attr.h:13:
In file included from /home/mmm/devel/HTM/htm-community/nupic.cpp/build/ThirdParty/pybind11/pybind11-src/include/pybind11/cast.h:13:
In file included from /home/mmm/devel/HTM/htm-community/nupic.cpp/build/ThirdParty/pybind11/pybind11-src/include/pybind11/detail/../pytypes.h:12:
In file included from /home/mmm/devel/HTM/htm-community/nupic.cpp/build/ThirdParty/pybind11/pybind11-src/include/pybind11/detail/common.h:146:
In file included from /usr/bin/../lib/gcc/x86_64-linux-gnu/8/../../../../include/c++/8/memory:80:
/usr/bin/../lib/gcc/x86_64-linux-gnu/8/../../../../include/c++/8/bits/unique_ptr.h:81:2: error: delete called on 'nupic::SDR_MetricsHelper_' that is abstract but has non-virtual destructor
[-Werror,-Wdelete-non-virtual-dtor]
delete __ptr;
^
/usr/bin/../lib/gcc/x86_64-linux-gnu/8/../../../../include/c++/8/bits/unique_ptr.h:274:4: note: in instantiation of member function
'std::default_delete<nupic::SDR_MetricsHelper_>::operator()' requested here
get_deleter()(__ptr);
^
/home/mmm/devel/HTM/htm-community/nupic.cpp/build/ThirdParty/pybind11/pybind11-src/include/pybind11/pybind11.h:1319:40: note: in instantiation of member function
'std::unique_ptr<nupic::SDR_MetricsHelper_, std::default_delete<nupic::SDR_MetricsHelper_> >::~unique_ptr' requested here
v_h.holder<holder_type>().~holder_type();
^
/home/mmm/devel/HTM/htm-community/nupic.cpp/build/ThirdParty/pybind11/pybind11-src/include/pybind11/pybind11.h:1054:26: note: in instantiation of member function
'pybind11::class_<nupic::SDR_MetricsHelper_>::dealloc' requested here
record.dealloc = dealloc;
^
/home/mmm/devel/HTM/htm-community/nupic.cpp/bindings/py/cpp_src/bindings/algorithms/py_SDR_Metrics.cpp:36:40: note: in instantiation of function template specialization
'pybind11::class_<nupic::SDR_MetricsHelper_>::class_<>' requested here
py::class_<SDR_MetricsHelper_> py_Helper(m, "SDR_MetricsHelper_");
^
In file included from /home/mmm/devel/HTM/htm-community/nupic.cpp/bindings/py/cpp_src/bindings/algorithms/py_SDR_Metrics.cpp:22:
In file included from /home/mmm/devel/HTM/htm-community/nupic.cpp/build/ThirdParty/pybind11/pybind11-src/include/pybind11/pybind11.h:43:
In file included from /home/mmm/devel/HTM/htm-community/nupic.cpp/build/ThirdParty/pybind11/pybind11-src/include/pybind11/detail/../attr.h:13:
In file included from /home/mmm/devel/HTM/htm-community/nupic.cpp/build/ThirdParty/pybind11/pybind11-src/include/pybind11/cast.h:13:
In file included from /home/mmm/devel/HTM/htm-community/nupic.cpp/build/ThirdParty/pybind11/pybind11-src/include/pybind11/detail/../pytypes.h:12:
In file included from /home/mmm/devel/HTM/htm-community/nupic.cpp/build/ThirdParty/pybind11/pybind11-src/include/pybind11/detail/common.h:146:
In file included from /usr/bin/../lib/gcc/x86_64-linux-gnu/8/../../../../include/c++/8/memory:80:
/usr/bin/../lib/gcc/x86_64-linux-gnu/8/../../../../include/c++/8/bits/unique_ptr.h:81:2: error: delete called on non-final 'nupic::SDR_Sparsity' that has virtual functions but non-virtual
destructor [-Werror,-Wdelete-non-virtual-dtor]
delete __ptr;
^
/usr/bin/../lib/gcc/x86_64-linux-gnu/8/../../../../include/c++/8/bits/unique_ptr.h:274:4: note: in instantiation of member function 'std::default_delete<nupic::SDR_Sparsity>::operator()'
requested here
get_deleter()(__ptr);
^
/home/mmm/devel/HTM/htm-community/nupic.cpp/build/ThirdParty/pybind11/pybind11-src/include/pybind11/pybind11.h:1319:40: note: in instantiation of member function
'std::unique_ptr<nupic::SDR_Sparsity, std::default_delete<nupic::SDR_Sparsity> >::~unique_ptr' requested here
v_h.holder<holder_type>().~holder_type();
^
/home/mmm/devel/HTM/htm-community/nupic.cpp/build/ThirdParty/pybind11/pybind11-src/include/pybind11/pybind11.h:1054:26: note: in instantiation of member function
'pybind11::class_<nupic::SDR_Sparsity, nupic::SDR_MetricsHelper_>::dealloc' requested here
record.dealloc = dealloc;
^
/home/mmm/devel/HTM/htm-community/nupic.cpp/bindings/py/cpp_src/bindings/algorithms/py_SDR_Metrics.cpp:52:54: note: in instantiation of function template specialization
'pybind11::class_<nupic::SDR_Sparsity, nupic::SDR_MetricsHelper_>::class_<char [490]>' requested here
py::class_<SDR_Sparsity, SDR_MetricsHelper_> py_Sparsity(m, "SDR_Sparsity",
^
In file included from /home/mmm/devel/HTM/htm-community/nupic.cpp/bindings/py/cpp_src/bindings/algorithms/py_SDR_Metrics.cpp:22:
In file included from /home/mmm/devel/HTM/htm-community/nupic.cpp/build/ThirdParty/pybind11/pybind11-src/include/pybind11/pybind11.h:43:
In file included from /home/mmm/devel/HTM/htm-community/nupic.cpp/build/ThirdParty/pybind11/pybind11-src/include/pybind11/detail/../attr.h:13:
In file included from /home/mmm/devel/HTM/htm-community/nupic.cpp/build/ThirdParty/pybind11/pybind11-src/include/pybind11/cast.h:13:
In file included from /home/mmm/devel/HTM/htm-community/nupic.cpp/build/ThirdParty/pybind11/pybind11-src/include/pybind11/detail/../pytypes.h:12:
In file included from /home/mmm/devel/HTM/htm-community/nupic.cpp/build/ThirdParty/pybind11/pybind11-src/include/pybind11/detail/common.h:146:
In file included from /usr/bin/../lib/gcc/x86_64-linux-gnu/8/../../../../include/c++/8/memory:80:
/usr/bin/../lib/gcc/x86_64-linux-gnu/8/../../../../include/c++/8/bits/unique_ptr.h:81:2: error: delete called on non-final 'nupic::SDR_ActivationFrequency' that has virtual functions but
non-virtual destructor [-Werror,-Wdelete-non-virtual-dtor]
delete __ptr;
^
/usr/bin/../lib/gcc/x86_64-linux-gnu/8/../../../../include/c++/8/bits/unique_ptr.h:274:4: note: in instantiation of member function
'std::default_delete<nupic::SDR_ActivationFrequency>::operator()' requested here
get_deleter()(__ptr);
^
/home/mmm/devel/HTM/htm-community/nupic.cpp/build/ThirdParty/pybind11/pybind11-src/include/pybind11/pybind11.h:1319:40: note: in instantiation of member function
'std::unique_ptr<nupic::SDR_ActivationFrequency, std::default_delete<nupic::SDR_ActivationFrequency> >::~unique_ptr' requested here
v_h.holder<holder_type>().~holder_type();
^
/home/mmm/devel/HTM/htm-community/nupic.cpp/build/ThirdParty/pybind11/pybind11-src/include/pybind11/pybind11.h:1054:26: note: in instantiation of member function
'pybind11::class_<nupic::SDR_ActivationFrequency, nupic::SDR_MetricsHelper_>::dealloc' requested here
record.dealloc = dealloc;
^
/home/mmm/devel/HTM/htm-community/nupic.cpp/bindings/py/cpp_src/bindings/algorithms/py_SDR_Metrics.cpp:95:13: note: in instantiation of function template specialization
'pybind11::class_<nupic::SDR_ActivationFrequency, nupic::SDR_MetricsHelper_>::class_<char [731]>' requested here
py_ActivationFrequency(m, "SDR_ActivationFrequency",
^
```

- Interestingly, this does not replicate on CI. 
- SDR_MetricsHelper_ does provide `~SDR_MetricsHelper_()`, does it need to be wrapped in pybind or something?

--- a/src/nupic/utils/SdrMetrics.hpp
+++ b/src/nupic/utils/SdrMetrics.hpp
@@ -124,7 +124,7 @@ public:
         callback( data, 1.0f / std::min( period_, (UInt) ++samples_ ));
     }
 
-    ~SDR_MetricsHelper_() {
+    virtual ~SDR_MetricsHelper_() {
         deconstruct();
     }
 };

